tests/cpp/ir/frontend_type_inference_test.cpp

```diff
@@ -52,31 +53,36 @@ TEST(FrontendTypeInference, UnaryOp) {
   auto prog = std::make_unique<Program>(Arch::x64);
   prog->config.default_fp = PrimitiveType::f64;
   auto const_i16 = value<int16>(-(1 << 10));
-  const_i16->type_check(nullptr);
+
+  CompileConfig dummy_config;
```

Comment:
Why these changes?

Reply:
`UnaryOpExpression::type_check(CompileConfig*)` will attempt to obtain `real_matrix` and `default_fp` from the `config` . It will hit an assertion if nullptr is passed in. 

(But I did overkilled a few Expressions other than UnaryOpExpression, will have that fixed, thanks!)